src/postBackEnd.js

```diff
@@ -0,0 +1,173 @@
+      if (queriedDocs.empty == false) {
+          queriedDocs.forEach(singleDoc => {
+              return singleDoc.data().address;
+          })
+      } else {
+          console.log("No docs match");
+      }
+  });
+}
+
+export function setAddress(post_id, newAddress){
+  db.collection('post').doc(post_doc).update({
+      addres: newAddress
+  }).catch(function (error) {
+      // The document probably doesn't exist.
+      console.error("Error updating document: ", error);
+      return false
+  });
+  return true
+}
+
+// Thinking it might be better to simply do one search for every comment
+// item with a given post_id than a serach for a specific comment id for
+// every value in an array of comments.  Will talk with Branden about this
+
+// export function getComments(post_id){
+//   const collect = db.collection('post')//get wanted collection
+//   query = collect.where('post_id', '==', post_id)
+//   query.get().then(queriedDocs => {
+//       if (queriedDocs.empty == false) {
+//           queriedDocs.forEach(singleDoc => {
+//               return singleDoc.data().comments;
+//           })
+//       } else {
+//           console.log("No docs match");
+//       }
+//   });
+// }
+//
+// export function setComments(post_doc, newComment){
+//   db.collection('post').doc(post_doc).update({
+//       hometown: newHometown
+//   }).catch(function (error) {
+//       // The document probably doesn't exist.
+//       console.error("Error updating document: ", error);
+//       return false
+//   });
+//   return true
+// }
+
+
+// Not in database currently, needs to be added
+// export function getUserPic(post_id){
+//   const collect = db.collection('post')//get wanted collection
+//   query = collect.where('post_id', '==', post_id)
+//   query.get().then(queriedDocs => {
+//       if (queriedDocs.empty == false) {
+//           queriedDocs.forEach(singleDoc => {
+//               return singleDoc.data().pictures;
+//           })
+//       } else {
+//           console.log("No docs match");
+//       }
+//   });
+// }
+//
+// export function setUserPic(post_doc, newPicLoc){
+//   db.collection('post').doc(post_doc).update({
+//       pictures: newPicLoc
+//   }).catch(function (error) {
+//       // The document probably doesn't exist.
+//       console.error("Error updating document: ", error);
+//       return false
+//   });
+//   return true
+// }
+
+export function setPostID(post_doc, newPostID){
+  db.collection('post').doc(post_doc).update({
+      postID: newPostID
+  }).catch(function (error) {
+      // The document probably doesn't exist.
+      console.error("Error updating document: ", error);
+      return false
+  });
+  return true
+}
+
+export function getTimeStamp(post_id){
+  const collect = db.collection('post')//get wanted collection
+  query = collect.where('post_id', '==', post_id)
+  query.get().then(queriedDocs => {
+      if (queriedDocs.empty == false) {
+          queriedDocs.forEach(singleDoc => {

```

Comment:
Confused on why you are calling `forEach()` if there is only a single doc is being read from the database I would think it wouldn't work with a single doc since it's not a list. Does this work, have you tested it? Same for the rest of your getters.

Reply:
The forEach() was recommended mainly for safety reasons.  For normal queries you might have more than just one entry, such as querying for comments using a post_id.  In these cases there aren't more than one doc, but figured I'd add it in there just in case since it didn't harm anything